### ISSUE TITLE
azurerm_linux_web_app_slot does not change service plan with parent resource

### ISSUE THREAD

**Issue by mewt666** [hashicorp/terraform-provider-azurerm#19474]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.2

### AzureRM Provider Version

3.33.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_web_app_slot, azurerm_linux_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_web_app" "webapps" {
  name                = main_webapp
  location            = azurerm_resource_group.rg.location
  resource_group_name = azurerm_resource_group.rg.name
  service_plan_id           = azurerm_service_plan.serviceplans.id
  https_only                = true
  virtual_network_subnet_id = azurerm_subnet.Subnet.id
}

resource "azurerm_linux_web_app_slot" "webapps-slot" {
  name                      = "PreProd-Slot"
  app_service_id            = azurerm_linux_web_app.webapps.id
  https_only                = true
  virtual_network_subnet_id = azurerm_subnet.Subnet.id
}
```

### Debug Output/Panic Output

```shell
│ Error: updating Linux Web App Slot: (Slot Name "PreProd-Slot" / Site Name "app_name" / Resource Group "rg"): web.AppsClient#CreateOrUpdateSlot: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
│
│   with azurerm_linux_web_app_slot.webapps-slot[],
│   on app_services.tf line 115, in resource "azurerm_linux_web_app_slot" "webapps-slot":
│  115: resource "azurerm_linux_web_app_slot" "webapps-slot" {
│
│ updating Linux Web App Slot: (Slot Name "PreProd-Slot" / Site Name "app_name" / Resource Group "rg"): web.AppsClient#CreateOrUpdateSlot: Failure sending request: StatusCode=0 --
│ Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
```

### Expected Behaviour

Upon changing of the service_plan_id in the parent resource, it is expected that both the parent resource and the deployment slot are moved to the service plan defined. 

### Actual Behaviour

Only the parent web app is moved and the deployment slot remains in the original service plan. Given that in our config there is also vnet integration, the slot fails to bind to the new vnet assignment producing the previously given error.

### Steps to Reproduce

define resources as given above.
terraform apply
change service plan assignment in parent web app
terraform apply 
Parent app is moved to new service plan, slot remains in previous service plan.

**Comment by daveneeley** [comment]
This sounds like it might have been addressed by #19459

**Comment by rcskosir** [comment]
@mewt666, thank you for opening this issue. The issue linked by @daveneeley does appear like a duplicate and has been addressed by this PR #20184. Therefore, I am going to mark this issue as closed. 

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new/choose) following the template. Thank you!
